### ISSUE TITLE
Constant crashes on linux

### ISSUE THREAD

**Issue by femtomatic** [CloudCompare/CloudCompare#534]
## Current Behavior

CloudCompare crashes constantly for no apparent reasons. The crashes can happen right after the application launches or during point cloud manipulations, but usually I can't work more than 5 to 10 minutes. The crashes seem to happen when the mouse hovers over the main window's title bar or when the mouse leaves the application window. When I launch CloudCompare from the console I get a very brief error message after the crash, usually one of the two below:
```
*** Error in `CloudCompare': free(): invalid pointer: 0x0000000001823eb0 ***
[1]    23929 segmentation fault (core dumped)  CloudCompare
```
or 
```
*** Error in `CloudCompare': free(): invalid pointer: 0x0000000001458740 ***
*** Error in `CloudCompare': free(): invalid pointer: 0x0000000001398ad0 ***
CloudCompare: malloc.c:3728: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
[1]    24080 abort (core dumped)  CloudCompare
```
## Expected Behavior

The app should not crash...
## Steps to Reproduce (for bugs)

Start the application under archlinux and move the mouse around, leaving the app window with the mouse.
## Context/Comments

I compile CloudCompare directly from the master branch and this behavior has started appearing a month or so ago on the 2.8 version. I don't see any unusual errors or warnings during compilation.
## Your Environment

* OS & Version: Archlinux, kernel 4.11.6
* CC Version: cloudcompare-git 2.8.0.r158.gab5f340a-1
* Graphics card: GeForce GTX 970 with the nvidia 381.22 linux driver
* GCC version: 7.1.1
* QT version: 5.9.0
* CMake version: 3.8.2

**Comment by dgirardeau** [comment]
This looks like an issue with the UI management and therefore I would say Qt is the culprit...

Qt 5.9 has not been tested / evaluated on our side. Can you maybe compile with an older version? (5.7 is the one I used personally).

**Comment by femtomatic** [comment]
Thanks for your time @dgirardeau, unfortunately it would be too complicated to roll back the Qt version to 5.7 on my system, I would risk breaking other packages. If all else fails I might install CC in a virtual machine with a more "stable" linux distribution.

I have tried compiling CC with the `-DCMAKE_BUILD_TYPE=Debug` cmake option and the compilation fails at around 80% with a bunch of [errors related to liblas](https://pastebin.com/ka8dPEu0). I have liblas version 1.8.1, but this hasn't changed since last October so I don't know what is going on. 

Also, today CC will not even get to the main interface, it crashes right after the small splash screen appears with the same error as before:
```
CloudCompare: malloc.c:3728: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
[1]    2053 abort (core dumped)  CloudCompare
```
I hope this helps in any way but I'm afraid it's some weird conflict with my system setup because I've asked other archlinux users and they don't have this problem.

**Comment by dgirardeau** [comment]
I just tried Qt 5.9 on my side (on Windows) and it works fine.

Which means it's probably an issue specific to Linux (or your distribution?). You should probably look at similar issues involving Qt 5.9... Even though it might not help you a lot :D

**Comment by femtomatic** [comment]
Hi! OK sorry for the long silence, but I finally figured it out! 

So I almost went crazy over this... I reinstalled and recompiled every dependency at least 2 or 3 times, reset all the Qt and CC config files and nothing worked. I was trying to debug for the n-th time and for some reason decided to reset my custom font configuration file and it worked! No more crashes!

So after some more digging, it seems like CC doesn't like the [droid fonts](http://www.droidfonts.com/) for some reason, or anyways the version that is in the Archlinux repos. In the end I just use another font and everything works fine.

Anyways, I would just like to add that I really like CC, it's really impressive how smoothly it can work with millions of points. I'm still new to LiDAR data processing so I don't really know how it compares to other proprietary solutions, but I think CC looks very professional and is a great example of open source software done well. So thanks for your work @dgirardeau and I hope I'll be able to contribute in a more useful way in the future!
